modal/functions.py

```diff
@@ -1064,10 +1070,9 @@ async def _load(param_bound_func: _Function, resolver: Resolver, existing_object
 
         fun: _Function = _Function._from_loader(_load, "Function(parametrized)", hydrate_lazily=True)
 
-        if can_use_parent and self.is_hydrated:
-            # Edge case that lets us hydrate all objects right away
-            # if the instance didn't use explicit constructor arguments
-            fun._hydrate_from_other(self)
+        if can_use_parent and parent.is_hydrated:
```

Comment:
Keeping this for now since it breaks a test due to how our Resolver works (the _load methods requires setting up a resolver + client, so it can't short circuit that part from within the function)